pkg/transport/queuestorage/file.go

```diff
@@ -23,37 +45,34 @@ type FileWriter struct {
 
 	store  filestore.StorageBackend
 	queue  queue.QueueBackend
-	notify chan FileEvent
+	notify chan FileWriterInfo
 
 	maxFileSize int64
 	maxRows     int64
 	expiry      time.Time
 
-	// Current file being written to
+	// fd is the file descriptor of the target file
 	fd *os.File
 
-	// Ensure only 1 file write (or rotate) is happening at a time
+	// canWrite ensure a sequential file write operation
 	canWrite sync.Mutex
 
-	// Used to rotate every x interval
-	ticker *time.Ticker
-
-	wg sync.WaitGroup
+	// stopTicker signals for the stoppage of file age ticker
+	stopTicker chan struct{}
 }
 
-type NewFileWriterParam struct {
-	Key    string
-	Path   string
-	Store  filestore.StorageBackend
-	Queue  queue.QueueBackend
-	Notify chan FileEvent
+func NewFileWriter(param NewFileWriterParam) (*FileWriter, error) {
+	isZero := func(x any) bool {
+		return reflect.ValueOf(x).IsZero()
+	}
 
-	MaxFileSize int64
-	MaxRows     int64
-	MaxFileAge  time.Duration
-}
+	if isZero(param.MaxFileSize) {
+		param.MaxFileSize = MaxFileSize
```

Comment:
Reasonably defaults are ideal in many cases. It would be ideal to enforce it on parameters like `key`, `path`, etc.

Reply:
Let's think about defaults separately. For example, perhaps default values can live alongside the `config` package rather than here. If we use a more sophisticated config [library](https://github.com/spf13/viper) then we can have all our defaults in one place.